hazelcast/src/main/java/com/hazelcast/config/BTreeIndexConfig.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.config;
+
+import com.hazelcast.internal.config.ConfigDataSerializerHook;
+import com.hazelcast.memory.Capacity;
+import com.hazelcast.memory.MemoryUnit;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Configuration options specific for B-Tree indexes.
+ *
+ * @since 5.2
+ */
+public class BTreeIndexConfig implements IdentifiedDataSerializable {
+
+    /**
+     * Default value of page size for B-Trees.
+     */
+    public static final Capacity DEFAULT_PAGE_SIZE = Capacity.of(1024, MemoryUnit.BYTES);
```

Comment:
I think for t-store b+tree we should raise the default to something like 8-64kB at least, let's start with 16, performance testing will tell us a proper default later.

For the regular HD b+tree `GlobalIndexPoolingAllocator#DEFAULT_BTREE_INDEX_NODE_SIZE` is used as a default currently, which is 8kB. Would the new setting deprecate/override it? If yes, we probably should have different defaults for the regular and t-store b+trees.

BTW, looks like a natural unit here is kilobytes.